File: c/src/algo/arrange.c
/* ************************************************************************** */
/*                                                                            */
/*                                                        :::      ::::::::   */
/*   arrange.c                                          :+:      :+:    :+:   */
/*                                                    +:+ +:+         +:+     */
/*   By: vpolard <vpolard@student.42.fr>            +#+  +:+       +#+        */
/*                                                +#+#+#+#+#+   +#+           */
/*   Created: 2026/03/02 13:04:16 by vpolard           #+#    #+#             */
/*   Updated: 2026/03/09 12:04:19 by vpolard          ###   ########.fr       */
/*                                                                            */
/* ************************************************************************** */

#include "algo.h"

void	arrange(t_package *package)
{
	int	*s;

	s = package->a->content;
	if (s[0] > s[1] && s[1] < s[2] && s[0] < s[2])
		sa(package);
	else if (s[0] > s[1] && s[1] > s[2] && s[0] > s[2])
	{
		sa(package);
		rra(package);
	}
	else if (s[0] > s[1] && s[1] < s[2] && s[0] > s[2])
		ra(package);
	else if (s[0] < s[1] && s[1] > s[2] && s[0] < s[2])
	{
		sa(package);
		ra(package);
	}
	else if (s[0] < s[1] && s[1] > s[2] && s[0] > s[2])
		rra(package);
}

File: rust/src/algo/arrange.rs
use super::Package;

impl Package {
    pub fn arrange(&mut self) -> &mut Self {
        let top = self.a.content[0];
        let mid = self.a.content[1];
        let bot = self.a.content[2];
        if top > mid && mid < bot && top < bot {
            self.sa();
        } else if top > mid && mid > bot {
            self.sa();
            self.rra();
        } else if top > mid && mid < bot && top > bot {
            self.ra();
        } else if top < mid && mid > bot && top < bot {
            self.sa();
            self.ra();
        } else if top < mid && mid > bot && top > bot {
            self.rra();
        }
        self
    }
}
